File: Bachelor/Semester4/Systems_for_Design_and_Implementation/C#/ProiectCShMPP/Server/service/ReservationService.cs
using System;
using log4net;
using ProiectCShMPP.domain;
using ProiectCShMPP.repository;

namespace ProiectCShMPP.service
{
    public class ReservationService
    {
        private readonly IReservationRepository _repository;
        private static readonly ILog Logger = LogManager.GetLogger(typeof(ReservationService));

        public ReservationService(IReservationRepository repository)
        {
            _repository = repository;
        }

        /// <summary>
        /// Method for saving a reservation
        /// </summary>
        /// <param name="client">the client that made the reservation</param>
        /// <param name="phoneNumber">the client's phone number</param>
        /// <param name="tickets">the desired number of tickets</param>
        /// <param name="trip">the trip to book</param>
        /// <param name="user">the user that registers</param>
        /// <returns>
        /// - null, if the reservation is successfully saved
        /// - the reservation, otherwise
        /// </returns>
        public Reservation Save(String client, String phoneNumber, int tickets, Trip trip, User user)
        {
            Logger.InfoFormat("Entry Reservation Save {0} {1} {2} {3} {4}", client, phoneNumber, tickets, trip, user);
            Reservation reservation = new Reservation(1L)
            {
                Client = client,
                PhoneNumber = phoneNumber,
                Tickets = tickets,
                Trip = trip,
                User = user
            };
            var result = _repository.Save(reservation);
            Logger.InfoFormat("Exit Reservation Save {0}", result);
            return result;
        }
    }
}

File: Bachelor/Semester4/Systems_for_Design_and_Implementation/Java/Server/src/main/java/service/ReservationService.java
package service;

import domain.Reservation;
import domain.Trip;
import domain.User;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import repository.ReservationRepository;

import java.util.Optional;

public class ReservationService {
    private final ReservationRepository repository;
    private final static Logger logger = LogManager.getLogger();

    public ReservationService(ReservationRepository repository) {
        this.repository = repository;
    }

    /**
     * Method for saving a reservation
     * @param client: String, the client that made the reservation
     * @param phoneNumber: String, the client's phone number
     * @param tickets: int, the desired number of tickets
     * @param trip: Trip, the trip to book
     * @param user: User, the user that registers
     * @return an {@code Optional}
     *          - null, if the reservation is successfully saved
     *          - the reservation, otherwise
     */
    public Optional<Reservation> save(String client, String phoneNumber, int tickets, Trip trip, User user){
        logger.traceEntry();
        Reservation reservation = new Reservation(1L, client, phoneNumber, tickets, trip, user);
        var result = repository.save(reservation);
        logger.traceExit(result);
        return result;
    }
}
